.github/workflows/vtk-pre-test.yml

```diff
@@ -0,0 +1,74 @@
+name: VTK Master Testing
+
+on:
+  # TODO: remove
+  push:
+    branches:
+      - maint/vtk-master-testing
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  ALLOW_PLOTTING: true
+  SHELLOPTS: 'errexit:pipefail'
+
+jobs:
+  Linux:
+    name: Linux VTK Master Testing
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7, 3.8, 3.9, "3.10"]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: requirements_test.txt
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-1-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-1-
+
+      - name: Build wheel and install pyvista
+        run: |
+          pip install build
+          python -m build --wheel
+          pip install dist/pyvista*.whl
+
+      - name: Set up vtk
+        run: |
+          pip uninstall vtk --yes
+          pip install vtk --pre --no-cache --extra-index-url https://gitlab.kitware.com/api/v4/projects/13/packages/pypi/simple
+
+      - uses: awalsh128/cache-apt-pkgs-action@v1.1.3
+        with:
+          packages: libgl1-mesa-glx xvfb python-tk
+          version: 3.0
```

Comment:
Let's use the new OSMesa wheel variants when they get published later this/next week and remove libgl1-mesa-glx

Reply:
While we're here, why do we install `python-tk`?

Reply:
No idea... had the same thought. This could be an artifice from qt?

Reply:
Potentially.

Easier to ask forgiveness. Let's remove it and see.

Reply:
I'll do this in a separate PR that cleans up the other CI jobs and leverages the OSMesa wheel for the 9.2.5 release

(Also upgrading jobs like docs to 3.11 where possible)